src/Illuminate/Database/Eloquent/Relations/Concerns/InteractsWithPivotTable.php

```diff
@@ -355,26 +377,36 @@ protected function hasPivotColumn($column)
      */
     public function detach($ids = null, $touch = true)
     {
-        $query = $this->newPivotQuery();
+        if ($this->using) {
+            $results = 0;
+            foreach ($this->parseIds($ids) as $id) {
+                $results += $this->newPivot([
+                    $this->foreignPivotKey => $this->parent->getKey(),
+                    $this->relatedPivotKey => $id,
+                ], true)->delete();
```

Comment:
Same comment/suggestion regarding readability of chainged calls with long/some args.

Reply:
same response as above. 👆